src/sources/nats.rs

```diff
@@ -204,12 +216,64 @@ async fn create_subscription(
 mod tests {
     #![allow(clippy::print_stdout)] //tests
 
+    use lookup::{owned_value_path, LookupBuf};
+    use value::{kind::Collection, Kind};
+    use vector_core::schema::Definition;
+
     use super::*;
 
     #[test]
     fn generate_config() {
         crate::test_util::test_generate_config::<NatsSourceConfig>();
     }
+
+    #[test]
+    fn output_schema_definition_vector_namespace() {
+        let config = NatsSourceConfig {
+            log_namespace: Some(true),
+            ..Default::default()
+        };
+
+        let definition = config.outputs(LogNamespace::Vector)[0]
+            .clone()
+            .log_schema_definition
+            .unwrap();
+
+        let expected_definition =
+            Definition::new_with_default_metadata(Kind::bytes(), [LogNamespace::Vector])
+                .with_meaning(LookupBuf::root(), "message")
+                .with_metadata_field(&owned_value_path!("vector", "source_type"), Kind::bytes())
+                .with_metadata_field(
+                    &owned_value_path!("vector", "ingest_timestamp"),
+                    Kind::timestamp(),
+                );
+
+        assert_eq!(definition, expected_definition);
```

Comment:
I wonder if we should/need to try and test more generically across decoders, or across all decoders - not just the Bytes one. I'm not sure what exactly that would look like without a lot of repetitive typing but throwing it out there.

Reply:
Not sure I understand what you mean? This isn't using Bytes decoder?

Reply:
https://vector.dev/docs/reference/configuration/sources/nats/#decoding.codec <- defaults to bytes, therefore this is implicitly using the bytes deserializer. Which is clear to see given the `root()` is the "message", which happens when the deserializer is Bytes.